sktime/transformations/series/hurst.py

```diff
@@ -0,0 +1,224 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Hurst Exponent Transformer for time series analysis."""
+
+# __author__ = ["phoeenniixx"]
+
+import numpy as np
+import pandas as pd
+from typing import Union, List, Optional
+import matplotlib.pyplot as plt
+from scipy import stats
+
+from sktime.transformations.base import BaseTransformer
+
+
+class HurstExponentTransformer(BaseTransformer):
+    """Transformer for calculating the Hurst exponent of a time series.
+
+    This transformer calculates the Hurst exponent, which is used to evaluate
+    the autocorrelation properties of time series, particularly the degree of
+    long-range dependence.
+
+    Parameters
+    ----------
+    lags : Optional[Union[List[int], range]], default=None
+        The lags to use for calculation. If None, uses a range based on min_lag and max_lag.
+    method : str, default='rs'
+        The method to use for Hurst exponent calculation. Either 'rs' (rescaled range)
+        or 'dfa' (detrended fluctuation analysis).
+    min_lag : int, default=2
+        The minimum lag to use if lags is None.
+    max_lag : int, default=100
+        The maximum lag to use if lags is None.
+    fit_trend : str, default='c'
+        The trend component to include in the calculation.
+    confidence_level : float, default=0.95
+        The confidence level for the confidence interval calculation.
+
+    Attributes
+    ----------
+    hurst_estimate_ : float
+        The estimated Hurst exponent.
+    confidence_interval_ : tuple
+        The confidence interval for the Hurst exponent estimate.
+    """
+
+    _tags = {
+        "scitype:transform-input": "Series",
+        "scitype:transform-output": "Series",
+        "scitype:instancewise": True,
+        "scitype:transform-labels": "None",
+        "X_inner_mtype": "pd.Series",
+        "y_inner_mtype": "None",
+        "univariate-only": True,
+        "requires_y": False,
+        "fit_is_empty": False,
+        "capability:inverse_transform": False,
+        "capability:unequal_length": True,
+        "handles-missing-data": False,
+        "authors": ["phoeenniixx"],
+        "maintainers": ["fkiraly"],
```

Comment:
put your name or leave the `maintainers` out, then it will default to `"sktime developers"` :-)